js/tests/unit/modal.js

```diff
@@ -976,4 +976,42 @@ $(function () {
         backdrop: 'static'
       })
   })
+
+  QUnit.test('should get modal-static class when clicking outside of modal-content if backdrop = static', function (assert) {
+    assert.expect(1)
+    var done = assert.async()
+    var $modal = $('<div class="modal" data-backdrop="static"><div class="modal-dialog" style="transition-duration: 20ms;"/></div>').appendTo('#qunit-fixture')
+
+    $modal.on('shown.bs.modal', function () {
+      $modal.trigger('click')
+      setTimeout(function () {
+        assert.ok($modal.hasClass('modal-static'), 'has modal-static class')
+        done()
+      }, 0)
+    })
+      .bootstrapModal({
+        backdrop: 'static'
+      })
+  })
+
+  QUnit.test('should not get modal-static class when clicking outside of modal-content if backdrop = static and event is prevented', function (assert) {
+    assert.expect(1)
+    var done = assert.async()
+    var $modal = $('<div class="modal" data-backdrop="static"><div class="modal-dialog" style="transition-duration: 20ms;"/></div>').appendTo('#qunit-fixture')
+
+    $modal.on('hidePrevented.bs.modal', function (e) {
+      e.preventDefault()
```

Comment:
Adding an assert here you should be good to ensure we pass here

Reply:
Not sure what you mean by that?

Reply:
sorry it's due to my bad english 😅 
I think you should add an assert here, to be sure this listener is called

Reply:
Haha ok got it. Done!